processor/logdedupprocessor/counter.go

```diff
@@ -51,6 +54,9 @@ func (l *logAggregator) Export() plog.Logs {
 
 		scopeLogs := resourceLogs.ScopeLogs().AppendEmpty()
 		for _, lc := range resource.logCounters {
+			// Count deduped logs
+			l.telemetryBuilder.DedupeProcessorDeduplicated.Add(context.Background(), lc.count)
```

Comment:
Does every log consumed by this processor end up bumping one log counter by one? Or does this only happen if the log is a duplicate of another?

Reply:
It is a total count of the number of logs that are aggregated together. eg the first time a counter is created, it will have a count of 1. I agree it doesn't make sense to say it's a dedup count if we isn't dedup any logs 🙂

We could we could either check if the count is > 1, then increment the dedup counter by the counter-1.

Reply:
Updated log name to aggregated logs in [35e07ed](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/34581/commits/35e07ed3cd5e42f94ef439f922f471e6d7419180) - I think this is a little more clear and doesn't mean we have to alter the value.

Reply:
Correct me if I'm wrong but this appears to just be a count of the number of logs which were passed to the processor. In other words, this is a very generic metric that could apply to all processors. I know there have been a few failed attempts to standardize metrics across components, but we have an [open proposal](https://github.com/open-telemetry/opentelemetry-collector/issues/10708) for adding this same metric (and a few others) to all processors. I tend to think we should solve this more generically, unless there is something particular about this metric that I'm missing.

Reply:
Yeah, I think your proposal and what's in this PR are trying to achieve similar results. 

What I was trying to achieve was with this was to show how well the aggregation was working. For example, if 10 log records were aggregated into a single output log record, I wanted this metric to show how many logs had been combined. Maybe a counter is the wrong instrument to use here because as you've suggested, this is effectively just counting the number of records that were received.

Reply:
What about a histogram or exponential histogram? This might give a better view into the aggregation patterns. The idea would be that each data point represents one aggregated log, with the value being the number of logs aggregated into one. 

Examples: 

Not much aggregation
| 1 | 2-3 | 4-7 | 8-15 | 
| --- | --- | --- | --- |
| 100 | 10 | 1 | 0 |

Lots of aggregation
| 1 | 2-3 | 4-7 | 8-15 | 
| --- | --- | --- | --- |
| 5 | 10 | 20 | 40 |

Reply:
That makes a lot of sense, I've updated the metric to a histogram in [334eafd](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/34581/commits/334eafd2d4f257e0554c9ec7a11818a9fa896c1b). Thanks for the suggestion 👍🏻